### PR TITLE
Accessibility: Search results page

### PR DESCRIPTION
**Pull request by Arth101**
### Changes:
- Change buttons to use button tags (was h2).
- Add correct semantic html tags (main, sections).
- Change "default" grey buttons to have correct constrasts (AA level).
- Add a better visual feedback, both color and underline, to selected filters.

### CODE REVIEW

File: collections-online/app/styles/variables.scss [hunk 1]
```diff
@@ -4,7 +4,7 @@ $white: #fff !default;
 $topbar-color: #262626 !default;
 $gray-darker: $black !default;
 $gray-dark: #333 !default;
-$gray: #555 !default;
+$gray: #e0e0e0 !default;
 $gray-light: #999 !default;
 $gray-lighter: hsl(345, 3%, 88%) !default;
```
Comment (hypesystem) [inline collections-online/app/styles/variables.scss:9]: It seems that `$gray-light` is now darker than `$gray`? Are `$gray-light` and `$gray-lighter` in use anywhere?

File: collections-online/app/views/mixins/search-filter-section.pug [hunk 2]
```diff
@@ -17,7 +17,7 @@ mixin filterSection(displayStringSingular, displayStringPlural, field, type)
           //- The user has not filtered this field or has not selected this value
           if !filters[field] || filters[field].indexOf(bucket.key) === -1
             - availableBuckets++;
-            button.btn.btn-default.search-filter-sidebar__filter-btn(
+            button.btn.btn-default.btn-no-shadow(.search-filter-sidebar__filter-btn
```
Comment (hypesystem) [inline collections-online/app/views/mixins/search-filter-section.pug:20]: Is the parenthesis `(` here a typo?

File: collections-online/app/views/mixins/search-filter-section.pug [hunk 3]
```diff
@@ -2,7 +2,7 @@ include ../mixins/date-interval-range-picker
 
 mixin filterSection(displayStringSingular, displayStringPlural, field, type)
   - var dataId = field + '-filters'
-  h2.search-filter-sidebar__heading(tabindex="0" class='search-filter-sidebar__heading--' + dataId)
```
Comment (hypesystem) [inline collections-online/app/views/mixins/search-filter-section.pug:5]: Should they maybe be buttons? Or something else? Or maybe have a label to convey that they are clickable? (I'm unsure that it is *clear enough* that they are interactive as is.)

Comment (hypesystem) [reply collections-online/app/views/mixins/search-filter-section.pug:5]: Ah, wait, I understand what elements these are now. Are we sure they shouldn't be headings of some kind? (Maybe h1's?) They do functionally seem like headings to me...

File: collections-online/app/views/includes/filterbar.pug [hunk 4]
```diff
@@ -2,38 +2,38 @@ if config.features.filterSidebar
...
         +icon('view-thumb')
-        h2.search-filter-sidebar__heading.search-filter-sidebar__tab Liste
+        button.search-filter-sidebar__heading.search-filter-sidebar__tab Liste
       .filterbar--mobile__button.filterbar--mobile__button--map(data-action='show-view-map')
         +icon('location')
-        h2.search-filter-sidebar__heading.search-filter-sidebar__tab Kort
+        button.search-filter-sidebar__heading.search-filter-sidebar__tab Kort
     .filterbar--mobile__container--close
-      .filterbar--mobile__button.filterbar--mobile__button--close(data-action='close-filter-menu')
+      button.filterbar--mobile__button.filterbar--mobile__button--close(data-action='close-filter-menu', aria-label='close')
```
Comment (hypesystem) [inline collections-online/app/views/includes/filterbar.pug:36]: `close` is english. Should probably be `Luk filtreringsmenuen` or similar

File: collections-online/app/views/mixins/search-filter-section.pug (hunk 2 above)
Comment (Arth101) [reply collections-online/app/views/mixins/search-filter-section.pug:20]: Yep

File: collections-online/app/views/includes/filterbar.pug (hunk 4 above)
Comment (Arth101) [reply collections-online/app/views/includes/filterbar.pug]: True. I will fix :)

File: collections-online/app/views/mixins/search-filter-section.pug (hunk 3 above)
Comment (Arth101) [reply collections-online/app/views/mixins/search-filter-section.pug:5]: Maybe. I am not sure since they are toggle elements that expand something.

**Pull request merged**